### PR TITLE
Feature/36 admin invoice show update invoice status

### LINKED ISSUE #42: feature/36-admin-invoice-show-update-invoice-status
36. Admin Invoice Show Page: Update Invoice Status

As an admin     
When I visit an admin invoice show page (/admin/invoices/:invoice_id)
I see the invoice status is a select field
And I see that the invoice's current status is selected
When I click this select field,
Then I can select a new status for the Invoice,
And next to the select field I see a button to "Update Invoice Status"
When I click this button
I am taken back to the admin invoice show page
And I see that my Invoice's status has now been updated

### PR DESCRIPTION
**Pull request by AlexKiselich**
## Checklist before requesting a review
- [x] Verified Render deploy is functional
- [x] Pulled/merged main 
- [x] Tests for the changes have been added into spec files
- [x] Branch name follows our guidelines 
- [x] All tests running clean
- [x] Ran this in Rails S  - features are working properly
- [x] Test coverage is at 100% for models and features

## What changes or new behavior does this PR introduce (Bug fix, feature, update, etc)?
  So I added an invoices controller with an update action so I could use a form_with model (@invoice) to update the invoice 
  status. 
  Updated admin invoice show spec test data
  Added tests for US36
  Updated admin invoice show page with form for invoice status
## Bugs, Issues, etc:
 - [ ] check if there are known bugs or issue

### CODE REVIEW

File: spec/features/admin/invoices/show_spec.rb [hunk 1]
```diff
@@ -72,5 +72,40 @@
...
+
+      visit admin_invoice_path(@invoice_2)
+
+      expect(page).to have_content("Invoice Status:")
+      expect(page).to have_field('invoice_status', with: "#{@invoice_2.status}")
+    end
+
+    it "will allow the admin to change field and click 'Update Invoice Status' then redirects back to admin invoice show page" do
+      visit admin_invoice_path(@invoice_1)
+      save_and_open_page
```
Comment (aplasket) [inline spec/features/admin/invoices/show_spec.rb:90]: would you mind removing or commenting out the save_and_open_page plz

File: app/views/admin/invoices/show.html.erb [hunk 2]
```diff
@@ -1,5 +1,10 @@
-<p>Invoice ID: <%= @invoice.id %>
-<p>Invoice Status: <%= @invoice.status %>
+<h2>Invoice ID: <%= @invoice.id %></h2>
+
+ <p>Invoice Status: <%= form_with model: @invoice, local: true do |form| %></p>
+                    <%= form.collection_select :status, ["in progress", "completed", "cancelled"], :to_s, :titleize, selected: @invoice.status %>
+                    <%= form.submit "Update Invoice Status"%>
+                    <% end %>
+
```
Comment (aplasket) [inline app/views/admin/invoices/show.html.erb:7]: um WOAH this is so freaking cool!! great job wiht this!! had to look up how this collection_select status works, very cool.

**Review by aplasket — APPROVED** [review]
UM WOAH. you're amazing. great job at this

File: app/views/admin/invoices/show.html.erb (hunk 2 above)
Comment (kbergstrom78) [reply app/views/admin/invoices/show.html.erb:7]: I would like to :ize everything.  :makethisworkilize, :testwillpassalize, etc.  Nice job!

**Review by kbergstrom78 — APPROVED** [review]

**Pull request merged**